oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/web/OAuth2LoginAuthenticationFilterTests.java

```diff
@@ -263,6 +267,23 @@ public void doFilterWhenCustomFilterProcessesUrlThenFilterProcesses() throws Exc
 		verify(this.filter).attemptAuthentication(any(HttpServletRequest.class), any(HttpServletResponse.class));
 	}
 
+	@Test
+	public void attemptAuthenticationWhenOAuth2AuthorizationRequestIsNullDoesNotCreateOAuth2AuthorizationResponse() throws IOException, ServletException {
```

Comment:
This test is pretty much the same as `doFilterWhenAuthorizationResponseAuthorizationRequestNotFoundThenAuthorizationRequestNotFoundError`.
It doesn't actually test the change you added.

Reply:
@jgrandja This is testing the invocation of `convert()` which creates an `OAuth2AuthorizationResponse`. If you revert the change, this will fail. So I think this is effectively testing the change.

I couldn't see any better way to test this. Any suggestion?

Reply:
@izeye You can mock private methods using [Powermock](https://github.com/powermock/powermock/wiki/Mockito#a-full-example-of-partial-mocking-of-a-private-method) and verify the zero interaction on the invocation.

Take a look at [OidcAuthorizationCodeAuthenticationProviderTests](https://github.com/spring-projects/spring-security/blob/master/oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/oidc/authentication/OidcAuthorizationCodeAuthenticationProviderTests.java#L412) which is mocking a private method.